tools/gotidy/main.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"regexp"
+
+	"github.com/rs/zerolog"
+	"github.com/sergi/go-diff/diffmatchpatch"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/tools/gotidy/githelpers"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/clihelper"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/osutil"
+)
+
+var gomodRegex = regexp.MustCompile("^go.mod$")
+
+type GoProject struct {
+	BeforeMod string
+	BeforeSum string
+	AfterMod  string
+	AfterSum  string
+}
+
+func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+	go func() {
+		<-ctx.Done()
+		stop() // restore default exit behavior
+		log.Println("Cancelling... interrupt again to exit")
+	}()
+	project := flag.String("path", ".", "Path to the go project to check for tidy.")
+	commit := flag.Bool("commit", false, "Commit the changes if there are any.")
+	subProjects := flag.Bool("subprojects", false, "Check subprojects for tidy.")
+	verbose := flag.Bool("v", false, "Print verbose output.")
+	flag.Parse()
+	Main(*project, *commit, *subProjects, *verbose)
+}
+
+// Main is the entrypoint for the gotidy tool
+func Main(project string, commit, subProjects, verbose bool) {
+	// clean up the project path
+	projectPath, err := filepath.Abs(filepath.Dir(project))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
would it make sense to wrap that error and explain what exactly has failed? i.e. that it was the project path?